pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,334 @@
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+
+    def wait_for_voltage_output_set(self, check_period=1.0, timeout=60.0):
+        """
+        Waits until HV voltage output reaches the voltage ouput setting.
+        Checks the voltage output every check_period seconds and raises an exception
+        if the voltage output doesn't reach the voltage setting until the timeout time.
+        :param check_period: voltage output will be measured every check_period (seconds) time
+        :param timeout: time (seconds) give to the voltage output to reach the voltage setting
+        :return: None
+        :raises: Exception if the voltage output can't reach the voltage setting
+        before the timeout completes (seconds)
+        """
+        log.info("Executing the wait_for_voltage_output_set function.")
+
+        ref_time = time.time()
+        future_time = ref_time + timeout
+
+        voltage_output_setting = self.voltage
+        time.sleep(self.COMMAND_DELAY)
+        error = 0.02  # error of +-ten volts (0.01kV) to enter into the voltage stable window
+        voltage_output = self.measure_voltage
+        voltage_output_set = (voltage_output > (voltage_output_setting - error)) and \
+                             (voltage_output < (voltage_output_setting + error))
+
+        while not voltage_output_set:
+            actual_time = time.time()
+            log.debug(f"\tWaiting for voltage output set. "
+                      f"Reading output voltage every {check_period} seconds.\n"
+                      f"\tTimeout: {timeout} seconds. Elapsed time: "
+                      f"{round(actual_time - ref_time, ndigits=1)} seconds.")
+
+            time.sleep(check_period)  # wait for voltage output reaches the voltage output setting
+            voltage_output = self.measure_voltage
+            voltage_output_set = (voltage_output > (voltage_output_setting - error)) and \
+                                 (voltage_output < (voltage_output_setting + error))
+            log.debug("voltage_output_valid_range: "
+                      "[" + str(voltage_output_setting - error) +
+                      ", " + str(voltage_output_setting + error) + "]")
+            log.debug("voltage_output: " + str(voltage_output))
+            if actual_time > future_time:
+                raise Exception("Timeout for wait_for_voltage_output_set function")
+
+        # if you are here is because the voltage_setting
+        # has been reaches at the output of the HV Voltage source
+
+    # Constructor
+    def __init__(self,
+                 adapter,
+                 response_encoding="iso-8859-2",
+                 query_delay=0.2,
+                 timeout=5000,
+                 write_termination="\n",
+                 read_termination="",
+                 send_end=True,
+                 includeSCPI=False,
```

Comment:
Can you change these options (read/write_termination, send_end, includeSCPI) on the device? If the device always uses these configuration options (I guess you cannot activate SCPI, for example), I would remove them from the init parameters and just call them in super init (e.g. ` super().__init__(includeSCPI=False)`).
It seems unnecessary to offer options to the user (e.g. write_termination), which he cannot use, as the device always needs a specific value ("\n" here).

Is it the same with the response_encoding?

Reply:
Answering your questions.

read/write_termination and send_end cannot be changed. So As you mentioned, It would be better to remove it from the constructor.  

On the other hand, the instrument has two command sets: The Eurotest command set and the SCPI command set and they are mutually exclusive. For the moment I'm implemented only the ET command set but i would like to leave the door open to switch (in the future) from one to another. 

For the response_encoding I think is totally internal to the instrument, so I will remove it from the constructor.

Many thanks

Reply:
Good to know about those two command sets. I know the same thing for another instrument.
In those cases, I think it is best to make to different instruments, if you want to have both command sets. Just state it in the device description. 
So I recommend just to do the one, you are going to use (ET in this case).

Reply:
Done in commit. Thanks